### PR TITLE
Tests done

### LINKED ISSUE #1: Analyse Ekubo ETH/wstETH pool and suggest optimum LP range strategy
## Context
1. We plan to create a strkfarm managed ETH/wstETH strategy on Ekubo. We need data analysis of past data to suggest LP parameters to manage investment with max profit
2. End usage: We will be providing certain liquidity to this pool in the suggested range and we will also rebalance the position from time to time to maintain maximum fee output while keeping impermanent loss less

## Methodology
1. Divide total available liquidity and swaps data into two parts: 70-30. 70% being used for training and optimisation, 30% used for testing the result

## Expected output
1. Analyse the liquidity and swaps data of ETH/wstETH pool with best TVL
2. Suggest logic for configuring lower & upper ticks for adding liquidity along with conditions to trigger rebalance
3. Show strategy performance on the test data

_

> ⚠️ **Note:** **Overfitting must be avoided. Do whatever optimisations u want on training data, once u run test on test data, share the result as it is. Do not optimise on test data.**

_

### PR DESCRIPTION
**Pull request by ManvithaMolakala**
Closes #1 

Added tests for Ekubo math functions.

### CODE REVIEW

File: src/strategies/ekubo_base.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def run_simulation(self, liquidity_range, rebalance_range, train_swap_df, token1
         sqrt_lower_bound = liquidity_lower_bound**0.5
         sqrt_upper_bound = liquidity_upper_bound**0.5
         sqrt_price = initial_price**0.5
-        initial_liquidity = get_liquidity(token1_provided, sqrt_price, sqrt_lower_bound)
+        initial_liquidity = get_liquidity(10**30, token1_provided, sqrt_price, sqrt_lower_bound, sqrt_upper_bound)
```
Comment (akiraonstarknet) [inline src/strategies/ekubo_base.py:51]: Add comment on what this number is

File: src/utils/ekubo_math.py [hunk 2]
```diff
@@ -119,29 +140,18 @@ def get_capital_token1_terms_from_L(L, sqrt_price, sqrt_lower_bound, sqrt_upper_
...
+    return  x + y*(sqrt_price**2) # sqrt_price = y/x . y = x* sp^2
 
 def get_capital_in_token1_terms_fromy(y, sqrt_price, sqrt_lower_bound, sqrt_upper_bound):
-    L = y / (sqrt_price - sqrt_lower_bound)
-    sqrt_price = max(min(sqrt_price, sqrt_upper_bound), sqrt_lower_bound)     
-    x = L * (sqrt_upper_bound - sqrt_price) / (sqrt_price * sqrt_upper_bound)
-    capital_in_token1 = y + x*(sqrt_price**2) # price_ratio = token0/token1
-    return capital_in_token1
+    
+    L = get_liquidity(10**30, y, sqrt_price, sqrt_lower_bound, sqrt_upper_bound)
```
Comment (akiraonstarknet) [inline src/utils/ekubo_math.py:149]: Add comment

File: src/utils/ekubo_math.py [hunk 3]
```diff
@@ -154,11 +164,7 @@ def get_liquidity_from_token1_capital(eth_capital, sqrt_price, sqrt_lower_bound,
...
     """
-    # Ensure the price is within the specified bounds
-    sqrt_price = max(min(sqrt_price, sqrt_upper_bound), sqrt_lower_bound)
-    
-    # Calculate liquidity (L) based on the provided ETH capital
-    liquidity = eth_capital * (sqrt_price * sqrt_upper_bound) / (sqrt_upper_bound - sqrt_price)
-    
-    return liquidity
-
+    y = total_capital_token1_terms*sqrt_price**2/(1+sqrt_price**2)
```
Comment (akiraonstarknet) [inline src/utils/ekubo_math.py:167]: Doesnt seem right.

File: tests/test_ekubomath.py [hunk 4]
```diff
@@ -1,4 +1,126 @@
...
+        3064071012072246010 * bool_to_sign(0)  # Expected result is 0 #x 
+    ),
+    (
+        1498166625277028332296 * bool_to_sign(1), #L
+        math.sqrt(1.000001 ** (158574 * bool_to_sign(0))), # sp
+        math.sqrt(1.000001 ** (158580 * bool_to_sign(0))), # sa
+        math.sqrt(1.000001 ** (158840 * bool_to_sign(0))), # sb
+        0  # Expected result is 0 #y
+    )
+]
```
Comment (akiraonstarknet) [inline tests/test_ekubomath.py:111]: No tests written for get_capital_token1_terms_from_L, get_capital_in_token1_terms_fromy, get_liquidity_from_token1_capital, impermanent_loss_absolute

**Review by akiraonstarknet — CHANGES_REQUESTED** [review]
